### PR TITLE
7.POST/api/articles/article_id/comments - Ready for Review

### PR DESCRIPTION
**Pull request by bobbynagra92**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -158,25 +158,81 @@ describe('/api/articles/:article_id/comments', () => {
...
+          comment_id: 19,
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          article_id: 1,
+        });
+      });
+  });
+  test('ERROR 400: Responds with an error message when attempting to POST an invalid comment', () => {
```
Comment (callumu21) [inline __tests__/app.test.js:202]: Make sure to keep this test description tied to the specific condition you're changing (e.g. when attempting to post with an invalid body property)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -158,25 +158,81 @@ describe('/api/articles/:article_id/comments', () => {
...
+       username: 'lurker',
+       body: 'test',
+     };
+     return request(app)
+       .post('/api/articles/not-a-num/comments')
+       .send(newComment)
+       .expect(400)
+       .then(({ body }) => {
+         expect(body.msg).toBe('Invalid input');
+       });
```
Comment (callumu21) [inline __tests__/app.test.js:236]: Just missing a few more tests here:

1. What happens when a comment body has a username that isn't in our system?
2. What happens when a comment body contains valid properties (username and body), as well as an extra property? (This should still be a successful request)
3. What happens if we try to post a comment to a valid review_id that doesn't exist in the system

File: models/articles.model.js [hunk 3]
```diff
@@ -42,4 +42,16 @@ exports.fetchArticleComments = (article_id) => {
     .then((result) => {
       return result.rows;
     });
+};
+
+exports.addComment = (article_id, comment) => {
```
Comment (callumu21) [inline models/articles.model.js:47]: Nice and readable model

**Review by callumu21 — CHANGES_REQUESTED** [review]
Really nice code so far! Just missing a few extra tests that I've put in a comment below. Once you've added those in, please resubmit this PR for review 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -158,25 +158,129 @@ describe('/api/articles/:article_id/comments', () => {
...
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Invalid input');
+      });
+  });
+  test('ERROR 400: Responds with an error message when attempting to POST an invalid username not yet in our database', () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:237]: This one supposed to be a 404 'Not found' error. Please refactor it.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -158,25 +158,129 @@ describe('/api/articles/:article_id/comments', () => {
...
+    const newComment = {
+      username: 'bobby',
+      body: 'test',
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(newComment)
+      .expect(400)
```
Comment (B7Bprog) [inline __tests__/app.test.js:245]: Expect 404 here.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -158,25 +158,129 @@ describe('/api/articles/:article_id/comments', () => {
...
+     };
+     return request(app)
+       .post('/api/articles/not-a-num/comments')
+       .send(newComment)
+       .expect(400)
+       .then(({ body }) => {
+         expect(body.msg).toBe('Invalid input');
+       });
+  });
+  test('ERROR 400: Responds with an error message when passed an article_id which does not yet exist', () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:273]: This should be a 404 'Not found'.

File: __tests__/app.test.js [hunk 7]
```diff
@@ -158,25 +158,129 @@ describe('/api/articles/:article_id/comments', () => {
...
+    const newComment = {
+      username: 'lurker',
+      body: 'test',
+    };
+    return request(app)
+      .post('/api/articles/99999/comments')
+      .send(newComment)
+      .expect(400)
```
Comment (B7Bprog) [inline __tests__/app.test.js:281]: Expect 404 here.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Bobby, please see my comments and refactor your tests, then send us a quick PR with it.

**Review by callumu21 — APPROVED** [review]
Nice work, Bobby! Looks good to merge 👍

**Pull request merged**